json/src/fs2/data/json/selectors.scala

```diff
@@ -28,16 +28,18 @@ object Selector {
   /** Selects the value in the object for which the key respects the predicate.
     * If the currently pointed value is not an object and `strict` is `true`,
     * then an error is raised, otherwise the value is skipped.
+    * If the currently pointed value is an object and `mandatory` is `true`,
+    * then an error is raised if the object doesn't contain all the names.
     */
-  case class NameSelector(pred: NamePredicate, strict: Boolean) extends Selector
+  case class NameSelector(pred: NamePredicate, strict: Boolean, mandatory: Boolean) extends Selector
   object NameSelector {
-    def apply(name: String, strict: Boolean) =
-      new NameSelector(NamePredicate.Single(name), strict)
-    def apply(names: Seq[String], strict: Boolean) =
+    def apply(name: String, strict: Boolean, mandatory: Boolean) =
+      new NameSelector(NamePredicate.Single(name), strict, mandatory)
+    def apply(names: Seq[String], strict: Boolean, mandatory: Boolean) =
       names match {
-        case Seq(name) => new NameSelector(NamePredicate.Single(name), strict)
-        case Seq()     => new NameSelector(NamePredicate.None, strict)
-        case _         => new NameSelector(NamePredicate.Several(names.toSet), strict)
+        case Seq(name) => new NameSelector(NamePredicate.Single(name), strict, mandatory)
+        case Seq()     => new NameSelector(NamePredicate.None, strict, mandatory)
+        case _         => new NameSelector(NamePredicate.Several(names.toSet), strict, mandatory)
```

Comment:
Is there a reason why they get parsed to a `List`, but converted to a `Set` here? Maybe parse the names directly to a `Set`?

Reply:
Probably no good beyond me being used to aggregate with `::`. I will build the set directly since order doesn't matter.